src/rust/engine/src/nodes.rs

```diff
@@ -644,37 +673,104 @@ impl From<Scandir> for NodeKey {
 
 ///
 /// A Node that captures an fs::Snapshot for a PathGlobs subject.
+/// TODO: ???
 ///
+#[derive(Clone, Eq, Hash, PartialEq)]
+pub struct Snapshot {
+  key: Key,
+  request_spec: SnapshotNodeRequestSpec,
+}
+
 #[derive(Clone, Debug, Eq, Hash, PartialEq)]
-pub struct Snapshot(Key);
+pub enum SnapshotNodeRequestSpec {
+  JustSnapshot,
+  WithGlobMatchData,
+}
+
+impl fmt::Debug for Snapshot {
+  fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+    write!(
+      f,
+      "Snapshot(key={}, request_spec={:?})",
+      self.key.to_string(),
+      self.request_spec
+    )
+  }
+}
+
+#[derive(Clone, Debug)]
+pub struct SnapshotExpansionResult {
+  snapshot: fs::Snapshot,
+  found_files: HashMap<PathGlob, GlobMatch>,
```

Comment:
Perhaps I'm missing something, but why would we expose the `found_files` out of VFS::expand rather than just eagerly failing in that method based on the configured behaviour?